tests/drivers/device_uid/api/src/main.c

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2019 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <device_uid.h>
+#include <ztest.h>
+#include <strings.h>
+
+/*
+ * @addtogroup t_device_uid_api
+ * @{
+ * @defgroup t_device_uid_get test_device_uid_get
+ * @brief TestPurpose: verify device uid get works
+ * @details
+ * - Test Steps
+ *   -# Read the UID
+ *   -# Check if to many bytes are written to the buffer
+ *   -# Check if UID is plausible
+ * - Expected Results
+ *   -# Device uid with correct length should be written to the buffer.
+ * @}
+ */
+
+#define BUFFER_LENGTH 17
+#define BUFFER_CANARY 0xFF
+
+#if defined(CONFIG_UID_STM32) || defined(UID_NRF)
```

Comment:
We shouldn't need this ifdef here, the sample/test should be agnostic to the driver implementation.